parl/algorithms/torch/maddpg.py

```diff
@@ -107,9 +78,30 @@ def __init__(self,
         self.critic_optimizer = torch.optim.Adam(
             lr=self.critic_lr, params=self.model.get_critic_params())
 
-    def predict(self, obs, use_target_model=False):
+    def predict(self, obs):
         """ use the policy model to predict actions
         
+        Args:
+            obs (torch tensor): observation, shape([B] + shape of obs_n[agent_index])
+
+        Returns:
+            act (torch tensor): action, shape([B] + shape of act_n[agent_index]),
+                noted that in the discrete case we take the argmax along the last axis as action
+        """
+        policy = self.model.policy(obs)
+
+        if self.continuous_actions:
+            action = policy[0]
+        else:
+            action = F.softmax(policy, dim=-1)
+        if self.continuous_actions:
+            action = torch.tanh(action)
```

Comment:
Why didn't you just place this line after line 94 ?

Reply:
modified